cli/src/cli_util.rs

```diff
@@ -2658,21 +2658,27 @@ fn resolve_default_command(
 
     if let Some(matches) = matches {
         if matches.subcommand_name().is_none() {
-            if config.get_string("ui.default-command").is_err() {
-                writeln!(
-                    ui.hint(),
-                    "Hint: Use `jj -h` for a list of available commands."
-                )?;
-                writeln!(
-                    ui.hint(),
-                    "Set the config `ui.default-command = \"log\"` to disable this message."
-                )?;
+            let mut default_command = match config.get("ui.default-command") {
```

Comment:
nit: `config.get(..).optional()?` can be used to report type error

Reply:
I saw the `.optional()?` method as developing this PR, but I didn't see how I could use it to make the code simpler here. Since here I need to report the hints whenever the command is not a `String` or `Vec` (aka `Structured` or `Err`), I don't see how adding the `Option` and the `?` (early return) would help.

Reply:
In order to handle invalid types (such as array of arrays), we'll need to look into the error variants. It's easier to map NotFound to Option and return the other errors by `?`.